src/aura/BGE_ConfigurationWizard/BGE_ConfigurationWizardHelper.js

```diff
@@ -1,1058 +1,611 @@
-                donationDateRange: '',
-                noMatchOnDate: false,
-                postProcessClass: ''
-            };
-        })(component, model);
+        //parse active fields because it's another level JSON object
+        let activeFields = JSON.parse(model.activeFields);
+        let allFields = model.availableFields;
+        this.loadAvailableFields(component, activeFields, allFields);
+
+        let availableFieldsBySObject = component.get('v.availableFieldsBySObject');
+        this.loadBatchFieldOptions(component, availableFieldsBySObject);
     },
 
     /**
-     * @description Returns the Batch Metadata View module.
-     * @param component. Lightning Component reference.
-     * @param model. The Model.
-     * @return View of the Batch Metadata module.
+     * @description: sets basic batch info and processing settings
+     * @param model: parsed JSON model from the model
      */
-    BatchMetadataView : function(component, model) {
-        return (function (component, model) {
-
-            // Subscribe to the model onMetadataUpdated event.
-            model.getBatchMetadata().onMetadataUpdated.subscribe(function() {
-                var batchMetadataView = component.get('v.batchMetadata');
-                var batchMetadata = model.getBatchMetadata();
-                var headerChanged = Boolean(batchMetadataView.pageHeader !== batchMetadata.pageHeader);
-
-                batchMetadataView.labels = batchMetadata.labels;
-                batchMetadataView.mode = batchMetadata.mode;
-                batchMetadataView.hasError = batchMetadata.hasError;
-                batchMetadataView.errorMessage = batchMetadata.errorMessage;
-                batchMetadataView.pageHeader = batchMetadata.pageHeader;
-                batchMetadataView.pendingSave = batchMetadata.pendingSave;
-                batchMetadataView.showAdvancedOptions = batchMetadata.showAdvancedOptions;
-
-                if (!batchMetadataView.hasError) {
-                    batchMetadataView.progressIndicatorStep = batchMetadata.progressIndicatorStep;
-                    _sendMessage('setStep',batchMetadata.progressIndicatorStep);
-                } else {
-                    component.find('notifLib').showNotice({
-                        'variant': 'error',
-                        'header': $A.get('$Label.c.PageMessagesError'),
-                        'message': batchMetadataView.errorMessage,
-                        closeCallback: function() {
-                            //callback action here
-                        }
-                    });
-                }
+    loadBatchInfo: function(component, model) {
+        let batchInfo = {};
+
+        //generic batch info
+        batchInfo.name = model.name;
+        batchInfo.id = model.id;
+        batchInfo.description = model.description;
+        batchInfo.expectedCount = (model.expectedCount === null || model.expectedCount === '') ? 0 : model.expectedCount;
```

Comment:
We could shorthand this as `batchInfo.expectedCount = modelExpectedCount || 0;` (And similar on next line)